Tests/MongoSwiftTests/ReadWriteConcernTests.swift

```diff
@@ -70,213 +71,330 @@ final class ReadWriteConcernTests: MongoSwiftTestCase {
                 .to(throwError(UserError.invalidArgumentError(message: "")))
     }
 
+    /// Checks that a `MongoClient` as well as the underlying mongoc clients have the expected read concern.
+    func checkClientReadConcern(_ client: MongoClient,
```

Comment:
These could probably be condensed into a single `checkConcern` that takes in the client/db/collection read/write concern as well as a closure to extract the temporary one.

e.g.
```swift 
func checkConcern<T: Equatable>(topLevel: T, expected: T, derived: () -> T, desc: String) -> Bool
```

Then for client e.g.
```swift
let clientRCDerivation = { client in
    return { 
        try client.connectionPool.withConnection { conn in
            let connRC = ReadConcern(from: mongoc_client_get_read_concern(conn.clientHandle))
            expect(connRC).to(equal(expected), description: description)
     }
}

checkConcern(client.readConcern, empty, clientRCDerivation(client), "client created with no RC provided") 
```
Totally up to you whether you find this approach preferable. I know in general I have no aversion to code duplication in tests, especially if it makes the test clearer / simpler, so this might not be worth doing. It just seemed like a fun/functional way to do it 😄

Reply:
I ended up condensing things a bit using a somewhat similar approach. the main difference is rather than using closures, the methods for getting RCs/WCs for pointers are protocol methods. this saves me having to pass in a closure every time I call the compare method.

Reply:
Nice, much more swifty.